src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/clinicaltrialparticipation/ClinicalTrialParticipationEvaluationConverter.java

```diff
@@ -0,0 +1,109 @@
+                break;
+            case "373067005":
+                geccoStudienteilnahmeEvaluation.setBereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode(BereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode.NO_QUALIFIER_VALUE);
+                break;
+            case "261665006":
+                geccoStudienteilnahmeEvaluation.setBereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode(BereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode.UNKNOWN_QUALIFIER_VALUE);
+                break;
+            case "74964007":
+                geccoStudienteilnahmeEvaluation.setBereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode(BereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode.OTHER_QUALIFIER_VALUE);
+                break;
+            case "385432009":
+                geccoStudienteilnahmeEvaluation.setBereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode(BereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode.NOT_APPLICABLE_QUALIFIER_VALUE);
+                break;
+            default:
+                throw new UnprocessableEntityException("Value code " + resource.getValueCodeableConcept().getCoding().get(0).getCode() + " is not supported");
+        }
+
+        if(geccoStudienteilnahmeEvaluation.getBereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode().equals(BereitsAnInterventionellenKlinischenStudienTeilgenommenDefiningCode.YES_QUALIFIER_VALUE)){
+
+            geccoStudienteilnahmeEvaluation.setStudienteilnahme(createStudyCluster(resource));
+        }
+
+        return geccoStudienteilnahmeEvaluation;
+    }
+
+    private StudienteilnahmeCluster createStudyCluster(Observation resource){
+
+        StudienteilnahmeCluster studienteilnahmeCluster = new StudienteilnahmeCluster();
+
+        StudiePruefungCluster studiePruefungCluster = new StudiePruefungCluster();
+        studienteilnahmeCluster.setStudiePruefung(studiePruefungCluster);
+
+        studiePruefungCluster.setTitelDerStudiePruefungDefiningCode(TitelDerStudiePruefungDefiningCode.PARTICIPATION_IN_INTERVENTIONAL_CLINICAL_TRIALS);
+
+        if(resource.getCode().hasText()){
+            studiePruefungCluster.setBeschreibungValue(resource.getCode().getText());
+        }
+
+        if(resource.hasComponent()){
+            studiePruefungCluster.setRegistrierung(createRegistryCluster(resource));
+        }
+
+        return studienteilnahmeCluster;
+    }
+
+    private List<StudiePruefungRegistrierungCluster> createRegistryCluster(Observation resource) {
+
+        StudiePruefungRegistrierungCluster studiePruefungRegistrierungCluster = new StudiePruefungRegistrierungCluster();
+
+        for (Observation.ObservationComponentComponent observationComponent
+                : resource.getComponent()) {
+
+            if(observationComponent.getCode().getCoding().get(0).getCode().equals("04")) {
```

Comment:
i would also put this maybe into an own method to increase readability

Reply:
erledigt